### PR TITLE
feat(ff-dashboard-templates): feature flag dashboard templates model

### PR DESCRIPTION
**Pull request by EDsCODE**
## Problem

- want feature flags to have related dashboards that can be created from templates

## Changes

- update feature flag model to have a many-to-many relationship with dashboards

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/api/feature_flag.py [hunk 1]
```diff
@@ -59,6 +61,11 @@ class FeatureFlagSerializer(TaggedItemSerializerMixin, serializers.HyperlinkedMo
     experiment_set: serializers.PrimaryKeyRelatedField = serializers.PrimaryKeyRelatedField(many=True, read_only=True)
     features: serializers.SerializerMethodField = serializers.SerializerMethodField()
     usage_dashboard: serializers.PrimaryKeyRelatedField = serializers.PrimaryKeyRelatedField(read_only=True)
+    dashboards = serializers.PrimaryKeyRelatedField(
```
Comment (neilkakkar) [inline posthog/api/feature_flag.py:64]: keeping up with the theme^

```suggestion
    analytics_dashboards = serializers.PrimaryKeyRelatedField(
```

File: posthog/api/test/test_feature_flag.py [hunk 2]
```diff
@@ -938,7 +943,7 @@ def test_my_flags_is_not_nplus1(self) -> None:
                 format="json",
             ).json()
 
-        with self.assertNumQueries(8):
+        with self.assertNumQueries(13):
```
Comment (neilkakkar) [inline posthog/api/test/test_feature_flag.py:946]: this looks like n+1 query?

File: posthog/api/test/test_feature_flag.py [hunk 3]
```diff
@@ -960,7 +965,7 @@ def test_getting_flags_is_not_nplus1(self) -> None:
                 format="json",
             ).json()
 
-        with self.assertNumQueries(9):
+        with self.assertNumQueries(15):
```
Comment (neilkakkar) [inline posthog/api/test/test_feature_flag.py:968]: same, n+1, not good, should be 11

File: posthog/api/feature_flag.py [hunk 4]
```diff
@@ -236,6 +244,13 @@ def update(self, instance: FeatureFlag, validated_data: Dict, *args: Any, **kwar
         if validated_key:
             FeatureFlag.objects.filter(key=validated_key, team=instance.team, deleted=True).delete()
         self._update_filters(validated_data)
+
+        dashboards = validated_data.pop("dashboards", None)
+
+        if dashboards is not None:
+            for dashboard in dashboards:
+                FeatureFlagDashboards.objects.create(dashboard=dashboard, feature_flag=instance)
```
Comment (neilkakkar) [inline posthog/api/feature_flag.py:252]: A test for when the dashboard already is in `FeatureFlagDashboards` please?

File: posthog/api/test/test_feature_flag.py (hunk 2 above)
Comment (EDsCODE) [reply posthog/api/test/test_feature_flag.py:946]: Ah good catch i meant to check this

File: posthog/api/test/test_feature_flag.py (hunk 3 above)
Comment (neilkakkar) [reply posthog/api/test/test_feature_flag.py]: whatever you think is best 😂 - I said 11 because the one above was 11, but paid no attention

**Review by neilkakkar — APPROVED** [review]

**Pull request merged**